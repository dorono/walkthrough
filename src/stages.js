export const STAGE_FACTOM = 'factom';
export const STAGE_ANCHORED = 'anchored';
export const STAGE_CONFIRMED = `${STAGE_FACTOM},${STAGE_ANCHORED}`;
export const STAGE_PENDING = 'replicated';

export const STAGE_PENDING_CHAIN_TEXT = 'This chain has not yet been written to the Factom Blockchain.';
export const STAGE_PENDING_ENTRY_TEXT = 'This entry has not yet been written to the Factom Blockchain.';
export const STAGE_PENDING_DATE = 'PENDING';

export const STAGE_NOT_AVAILABLE = 'Not yet available';
