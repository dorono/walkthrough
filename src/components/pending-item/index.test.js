import React from 'react';
import {shallow} from 'enzyme';
import {STAGE_PENDING} from 'stages';

import PendingItem from './index';

const defaultProps = {
    enableTooltip: true,
    stage: STAGE_PENDING,
};

describe('PendingItem Component', () => {
    it('should render without errors', () => {
        shallow(
            <PendingItem {...defaultProps} />,
        );
    });

    const wrapper = shallow(<PendingItem {...defaultProps} />);

    it('should display the tooltip on mouseover', () => {
        wrapper.instance().handleMouseEnter();
        expect(wrapper.state('displayTooltip')).toBe(true);
    });

    it('should hide the tooltip on mouseout', () => {
        wrapper.instance().handleMouseEnter();
        expect(wrapper.state('displayTooltip')).toBe(false);
    });

    it('should match snapshot', () => {
        expect(shallow(<PendingItem />)).toMatchSnapshot();
    });
});
