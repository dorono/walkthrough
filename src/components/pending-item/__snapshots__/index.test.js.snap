// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingItem Component should match snapshot 1`] = `
<div
  className="pendingContainer"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <Icon
    className="pendingItemIcon placeholderPadding"
    name="Schedule"
  />
  <Tooltip
    arrowDirection="left"
    autoHide={false}
    className="stateCallout"
    show={false}
    small={true}
    timeout={5000}
  >
    <span>
      PENDING
    </span>
  </Tooltip>
</div>
`;
